From 944f219f43a72a4d093a965a208ac4d8312f675b Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 04 Jan 2021 00:01:39 +0200
Subject: [PATCH] [SW-28128] gaudi: security: close lbw from sob

The Sync Object LBW access is not routed through the regular routers and
therefore does not abey their settings. Instead, it has its own set of
the RRs to be restricted by.
Close the LBW privilege areas access from Sync Object.

Change-Id: Ia59d9a11983beb514cf88ec10772574935760c6b
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index d5ab1e4..8fa4dd5 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -1064,7 +1064,11 @@
 
 enum preboot_sec_lbw_rr_type {
 	IF_RTR,
+	DMA_IF_SOB,
 };
+
+#define DMA_IF_NUM	4
+#define DMA_IF_DIST	0x20000
 
 struct preboot_sec_lbw_rrs_cfg preboot_sec_lbw_rrs_cfg[] = {
 	[IF_RTR] = {
@@ -1076,6 +1080,16 @@
 		.max_rd_breg = mmSIF_RTR_0_LBW_RANGE_PRIV_MAX_AR_0,
 		.hit_wr_breg = mmSIF_RTR_0_LBW_RANGE_PRIV_HIT_AW,
 		.hit_rd_breg = mmSIF_RTR_0_LBW_RANGE_PRIV_HIT_AR,
+	},
+	[DMA_IF_SOB] = {
+		.instances_num = DMA_IF_NUM,
+		.instances_dist = DMA_IF_DIST,
+		.min_wr_breg = mmDMA_IF_W_S_SOB_MIN_WPRIV_0,
+		.max_wr_breg = mmDMA_IF_W_S_SOB_MAX_WPRIV_0,
+		.min_rd_breg = mmDMA_IF_W_S_SOB_MIN_RPRIV_0,
+		.max_rd_breg = mmDMA_IF_W_S_SOB_MAX_RPRIV_0,
+		.hit_wr_breg = mmDMA_IF_W_S_SOB_HIT_WPRIV,
+		.hit_rd_breg = mmDMA_IF_W_S_SOB_HIT_RPRIV,
 	},
 };
 
@@ -1114,6 +1128,7 @@
 static void preboot_secure_lbw_rr(int rr, u64 min, u64 max)
 {
 	preboot_secure_lbw_rr_gen(rr, min, max, IF_RTR);
+	preboot_secure_lbw_rr_gen(rr, min, max, DMA_IF_SOB);
 	preboot_secure_pcie_rr(rr, min, max);
 }
 
